src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int DOUBLE_MULTIPLIER = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key);
+        if (size > 0 && replaceDuplicateKeyValue(key, value)) {
+            return;
+        }
+        table = (size == table.length * DEFAULT_LOAD_FACTOR) ? resize() : table;
+        Node<K, V> current = table[index];
+        Node<K, V> newNode = new Node<>(key, value, null);;
+        if (current != null) {
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        } else {
+            table[index] = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        if (!checkElementIndexExistence(key)) {
```

Comment:
checkElementIndexExistence must return something

Reply:
It returns boolean, so !checkElementIndexExistence(key) is the same as if (checkElementIndexExistence(key) == false)